### ISSUE TITLE
Locate user is not accurate and different in Chrome and Brave and not working in Firefox 

### ISSUE THREAD

**Issue by vin0010** [visgl/react-map-gl#1036]
```
import React, {useState} from 'react';
import 'mapbox-gl/dist/mapbox-gl.css'
import MapGL, {GeolocateControl} from 'react-map-gl';

const MAPBOX_TOKEN = 'MY_TOKEN';

const Map = () => {
    const [viewport, setViewport] = useState({

        latitude: 12.973319,
        longitude: 80.185742,
        zoom: 14,
        bearing: 0,
        pitch: 0
    });


    const geolocateStyle = {
        position: 'absolute',
        top: 0,
        left: 0,
        margin: 10
    };

    return (

        <MapGL
            {...viewport}
            width="100vw"
            height="100vh"
            mapStyle="mapbox://styles/mapbox/streets-v11"
            onViewportChange={setViewport}
            mapboxApiAccessToken={MAPBOX_TOKEN}
        >
            <GeolocateControl
                style={geolocateStyle}
                positionOptions={{enableHighAccuracy: true}}
                trackUserLocation={true}
            />
        </MapGL>
    );
}

export default Map;

```

**Outcome:**
Chrome:
Its not accurate, 3 blocks away
![image](https://user-images.githubusercontent.com/10086767/75118552-61cccf80-567b-11ea-843c-55161f875feb.png)

Firefox:
Not working
![image](https://user-images.githubusercontent.com/10086767/75118673-5a59f600-567c-11ea-94c8-ddb97f683ce2.png)

Brave:
7km away
![image](https://user-images.githubusercontent.com/10086767/75118686-87a6a400-567c-11ea-8fe0-ab52eb5228a6.png)

How can I fix this issue?

**Comment by vin0010** [comment]
Guys please let me know if you have any suggestions, since I am planning to use mapbox in prod for my application.

**Comment by yurivangeffen** [comment]
Without knowing much of the internals of this package, I would assume that this is more likely the result of differences in your browsers, and not related to this package (or even MapBox). Have you used any of the debugging tools the browser provides to see what's going on concerning the location?

**Comment by yurivangeffen** [comment]
I have similar differences between browsers in my application. Still, I think this is due to the browsers implementations. They might use different interfaces on the OS, which use different ways of figuring out the location (GPS, IP address, magic...). On iPhone Safari this works fine (as it uses GPS).

**Comment by ax00b3** [comment]
I encountered this issue last night. The problem is as follows: Firefox uses the Google Maps API for geolocation. However, Google limited/changed/broke Mozilla's access to the API so geolocation is currently broken in Firefox. If you go to `about.config` in Firefox and look for the `geo.provider.network.url` key, you can see that the current one is `https://www.googleapis.com/geolocation/v1/geolocate?key=%GOOGLE_LOCATION_SERVICE_API_KEY%`. This is currently broken and as far as I know, Mozilla have made very little efforts to fix it. To fix it on your own, you can replace the URL with `https://location.services.mozilla.com/v1/geolocate?key=test`. I do not know how long this will work and it has the inconvenience of having to ask the user to fiddle with their browser, but it will do in the meantime.

In the case of Chromium, it doesn't ship with a valid API key so things are broken too. Only Chrome and Brave come with valid API keys so your milage may vary.

so far, the lest intrusive solution I came up with was to issue fetch requests by myself and forgetting about the HTML5 geolocation API.

So TL;DR: Not a problem from react-map-gl but entirely because of the current state of browsers.
